MarsRover/RoverClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MarsRover
+{
+    public class RoverClient
+    {
+        private Grid _grid;
+        private IHeading _heading;
+        private List<Obstacle> _obstacles;
+        private Rover _rover;
+        private List<RoverNavigationHistory> _navigationHistory = new List<RoverNavigationHistory>();
+
+
+        public RoverClient()
+        {
+            _grid = new Grid(3, 3, 5, 5);
+            _heading = new NorthHeading();
+            _obstacles = CreateObstacles(3);
+            _rover = new Rover(_grid, _heading, _obstacles);
+            AddToNavigationHistory();
+        }
+
+        
+        public void RoverInstructions()
+        {
+            var commands = string.Empty;
+            DisplayInstructions();
+            
+            while(!commands.Equals("X"))
+            {
+                Console.WriteLine($"Your current position on the surface grid is {_rover.XCoordinate()}, {_rover.YCoodinate()} and the Rover is facing {_rover.Direction()}.\n");
+                Console.WriteLine("Please enter your rover commands: ");
+                commands = Console.ReadLine().ToUpper().Replace(" ", "");
+                _rover = _rover.ExecuteCommands(commands);
+
+                AddToNavigationHistory();
+                if (_rover.ObstacleDetected()) { ObstacleAvoidance(); } 
+            }
+        }
+
+
+        public List<Obstacle> CreateObstacles(int numberOfObstacles)
+        {
+            var obstacles = new List<Obstacle>();
+            var random = new Random();
+
+            for (var i = 0; i < numberOfObstacles; i++)
+            {
+                var obstacleXCoordinate = random.Next(0, _grid.GridWidth());
+                var obstacleYCoordinate = random.Next(0, _grid.GridHeight());
+
+                if (!(ObstacleSameAsStartingPoint(obstacleXCoordinate, obstacleYCoordinate)))
+                {
+                    var obstacle = new Obstacle(obstacleXCoordinate, obstacleYCoordinate);
+                    obstacles.Add(obstacle);
+                }
+            }
+            return obstacles;
+        }
+
+
+        public bool ObstacleSameAsStartingPoint(int x, int y) => (x == _grid.XCoordinate() && y == _grid.YCoordinate());
+
+
+        private void DisplayInstructions()
+        {
+            Console.WriteLine("****Mars Rover Controller****\n");
+            Console.WriteLine("Use the following letters to write a set of commands for the Rover to follow.  Invalid commands will be ignored:");
+            Console.WriteLine("         F = Forward     B = Backward");
+            Console.WriteLine("         L = Turn Left   R = Turn Right");
+            Console.WriteLine("            X = Exit the Controller\n");
+        }
+
+
+        private void ObstacleAvoidance()
```

Comment:
`ObstacleAvoidance` is a noun.  Functions should be verbs and classes nouns.  I would rename this to `AvoidObstacle`.  This will make where this is called read a bit nicer: `if (_rover.ObstacleDetected()) { AvoidObstacle(); } `

"if obstacle detected, avoid obstacle"

Reply:
This function was removed